spec/unit/null/page_spec.rb

```diff
@@ -20,4 +20,14 @@
     expect(pager.page(text)).to eq(text)
     expect(output.string).to eq('')
   end
+
+  it "doesn't write a newline if one wasn't given" do
+    allow(output).to receive(:tty?).and_return(true)
+    pager = described_class.new(output: output)
+    text = "I try all things, I achieve what I can."
+
+    pager.page(text)
+
+    expect(output.string).to eq(text)
+  end
```

Comment:
The reason for this change is because the `page` method doesn't use `puts` now, it uses `write`. Why is this different? Because (and this was kind of a surprise to me), if you `puts "something\n"` and `puts "something"`, those two produce the same results. So, `puts` doesn't *add* a newline, it ensures there is one at the end. I'm not sure if this was intentional, but there wasn't a test for this edge case, so here it is. It can be changed to "it writes a newline even if one wasn't given" and the code for `page` could use `puts` instead.

Reply:
I agree with your observation. Equally, why don't we add `puts` as a close sibling to `write` to the pagers?